Nemo Connect/Sources/NCNetworkSessionManager.m

```diff
@@ -207,4 +212,12 @@ - (void)URLSession:(NSURLSession *)session dataTask:(NSURLSessionDataTask *)data
     [delegate URLSession:session dataTask:dataTask didReceiveData:data];
 }
 
+- (void)URLSessionDidFinishEventsForBackgroundURLSession:(NSURLSession *)session
+{
+    if (self.backgroundTaskFinishedCompletionBlock)
+    {
+        self.backgroundTaskFinishedCompletionBlock();
```

Comment:
i can pass the URLSession, but it is not needed. The system will call this method when every operation finished in the session. In that point we have to call the completionBlock which is passed to the appdelegate of the host application. The session is not needed there, and the regular session methods are working (so the completion block will be called normally )